keras_contrib/applications/densenet.py

```diff
@@ -301,25 +373,93 @@ def __create_dense_net(nb_classes, img_input, include_top, depth=40, nb_dense_bl
-                               weight_decay=weight_decay)
-        nb_filter = int(nb_filter * compression)
 
-    # The last __dense_block does not have a __transition_block
-    x, nb_filter = __dense_block(x, nb_layers, nb_filter, growth_rate, bottleneck=bottleneck,
-                                 dropout_rate=dropout_rate, weight_decay=weight_decay)
+        x = BatchNormalization(mode=0, axis=concat_axis, gamma_regularizer=l2(weight_decay),
+                               beta_regularizer=l2(weight_decay))(x)
+        x = Activation('relu')(x)
+        x = GlobalAveragePooling2D()(x)
+        # image labeling version of final activation layer
+        if include_top:
+            x = Dense(nb_classes, activation='softmax', W_regularizer=l2(
+                weight_decay), b_regularizer=l2(weight_decay))(x)
+        return x
 
-    x = BatchNormalization(mode=0, axis=concat_axis, gamma_regularizer=l2(weight_decay),
-                           beta_regularizer=l2(weight_decay))(x)
-    x = Activation('relu')(x)
-    x = GlobalAveragePooling2D()(x)
+    else:
+        # perform image segmentation using FCN densenet
+        skip_connection = x
+        skip_list = []
+        # Add dense blocks and transition down block for image segmentation
+        for block_idx in range(nb_dense_block):
+            x, nb_filter = __dense_block(x, nb_dense_block_layers[block_idx], nb_filter, growth_rate, bottleneck=bottleneck,
+                                         dropout_rate=dropout_rate, weight_decay=weight_decay)
+            # Skip connection
+            x = merge([x, skip_connection], mode='concat',
+                      concat_axis=concat_axis)
+            skip_list.append(x)
+
+            # add transition_block
+            x = transition_down_block(x, nb_filter, compression=compression, dropout_rate=dropout_rate,
+                                      weight_decay=weight_decay)
+            # this is calculated inside transition_down_block
+            nb_filter = int(nb_filter * compression)
+
+            # Preserve transition for next skip connection after dense
+            skip_connection = x
+
+        # The last dense_block does not have a transition_down_block
+        x, nb_filter = dense_block(x, final_nb_layer, nb_filter, growth_rate, bottleneck=bottleneck,
+                                   dropout_rate=dropout_rate, weight_decay=weight_decay)
+
+        if K.image_dim_ordering() == 'th':
+            out_shape = [batch_size, nb_filter, rows // 16, cols // 16]
+        else:
+            out_shape = [batch_size, rows // 16, cols // 16, nb_filter]
+
+        # Add dense blocks and transition up block
+        for block_idx in range(nb_dense_block):
+            if K.image_dim_ordering() != 'th':
+                out_shape[3] = nb_filter
+
+            x = __transition_up_block(
+                x, nb_filters=nb_filter, type=upscaling_type, output_shape=out_shape)
```

Comment:
You removed the upscaling_type parameter in my original model definition, but still call it here.